### PR TITLE
P4-1481 complete move

### PR DESCRIPTION
**Pull request by martyn-w**
### Jira link

[P4-1481](https://dsdmoj.atlassian.net/browse/P4-1481)

~NB: currently based against `P4-1406-redirect-move-endpoint` for readability.~ PR https://github.com/ministryofjustice/hmpps-book-secure-move-api/pull/484 should be merged first. (This PR builds on #484, so will be simpler once it is merged.)

### What?

I have added/removed/altered:

- [x] Added `complete a move` event
- [x] Added supporting tests

### Why?

I am doing this because:

- so we can switch to an event-based approach for changing a move's status

### Have you? (optional)

- [X] Updated API docs if necessary	

### Deployment risks (optional)

- minimal - this is a new endpoint which no one is using yet

### CODE REVIEW

File: app/services/move_events/params_validator.rb [hunk 1]
```diff
@@ -14,6 +15,7 @@ class ParamsValidator
 
     def initialize(params)
       @timestamp = params.dig(:attributes, :timestamp)
+      @type = params.dig(:type)
```
Comment (smoothcontract) [inline app/services/move_events/params_validator.rb:18]: Looks like Rubocop will soon come looking for these single argument digs, so maybe replace with `params[:type]` ?

https://github.com/rubocop-hq/rubocop/issues/7949

File: spec/requests/api/v1/move_events_controller_complete_spec.rb [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    let(:move) { create(:move) }
+    let(:move_id) { move.id }
+    let(:new_location) { create(:location) }
+    let(:complete_params) do
+      {
+        data: {
+          type: 'complete',
+          attributes: {
+            timestamp: '2020-04-23T18:25:43.511Z',
+            notes: 'jobs a good un',
```
Comment (smoothcontract) [inline spec/requests/api/v1/move_events_controller_complete_spec.rb:26]: 👏

File: spec/services/event_log/move_runner_spec.rb [hunk 3]
```diff
@@ -5,55 +5,87 @@
...
+      expect(Notifier).not_to have_received(:prepare_notifications)
     end
   end
 
-  context 'when events are not received in a chronological order' do
-    let(:event1_timestamp) { '2020-05-22 10:00:00' } # chronological second event received first
-    let(:event2_timestamp) { '2020-05-22 09:00:00' } # chronological first event received last
+  context 'when event_name=redirect' do
+    let!(:event1) { create(:move_event, :redirect, eventable: move, client_timestamp: event1_timestamp, created_at: 100.minutes.ago, details: event1_details) }
+    let(:event1_details) { { note: 'Event1', event_params: { relationships: { to_location: { data: { id: event1_location.id } } } } } }
```
Comment (smoothcontract) [inline spec/services/event_log/move_runner_spec.rb:40]: These are a bit tricky to read, maybe break over multiple lines? Same on line 44.

Comment (martyn-w) [reply spec/services/event_log/move_runner_spec.rb]: 👍

File: app/services/move_events/params_validator.rb (hunk 1 above)
Comment (martyn-w) [reply app/services/move_events/params_validator.rb]: 👍

**Pull request merged**